10up/wp-scaffold
# Spatie Ignition for Debugging

### Description of the Change
This PR adds a new plugin to the scaffold. The plugin adds [`spatie/ignition`](https://github.com/spatie/ignition/) as a debugging tool. Ignition is the default error page for Laravel Projects, and for good reason. It includes great stack traces as well as other helpful information that can be used to debug the issue. There's an example of an issue page below:
![wp-scaffold test_](https://github.com/user-attachments/assets/5afa053e-1be9-42b3-934c-23b475c2d15d)

It lets engineers quickly jump to the problem code, reducing debugging time (see video below).

https://github.com/user-attachments/assets/73e82430-8837-4f82-819c-fc4a5e3cc299

We load it using the `000-wp-ignition-loader.php` file as it needs to run as soon as possible in order to catch errors in MU plugins.

Whilst this should be ignored from deployments to production environments, since we should be handling errors transparently there, it does have some protections in place to ensure it doesn't run, as long as the `wp_get_environment_type()` returns correctly.

> [!NOTE]
> This update doesn't handle Fatal Errors, like Abstract Class extension issues because they happen at compile-time and therefore are handled by PHP's and WordPress's `register_shutdown_function()` before this package has time to process them.

### How to test the Change
Clone locally. Change something in the plugins/theme to cause a PHP error. See Ignition.

### Changelog Entry
* Added - Ignition as an error page.

### Credits

Props @darylldoyle 

### Checklist:
